### DISCUSSION TITLE
Crashes of the settings app

### DISCUSSION THREAD

**Discussion by brewmark27** [seaky/nspanel_pro_tools_apk#64] [Q&A]
Hi,

I'm not sure whether it is related to the NSPanel pro tools, but I have a problem with my NS Panel. The settings app crashes when I open the battery settings or the advanced display settings. This problem occurs since updating to V2.0.1 of the NSPanel Pro Tools, and to the most recent version of the home assistant app. Anybody experiencing the same problem? I have tried restarting the device and disabling the NSPanel Pro Tools, but with no result.

Regards,
Mark

**Comment by seaky** [comment]
What is your "Display sleep" settings in the app?

**Reply by brewmark27** [reply]
after 30 minutes of inactivity

**Comment by seaky** [comment]
furthermore post here the log:
`adb shell "ps -A | grep com.android.settings"`
first number after system is the pid

`adb logcat --pid=<pid>`

**Reply by brewmark27** [reply]
Please find the log below:

--------- beginning of main
02-20 16:25:51.714 16622 16639 D BatteryInfo: time to make batteryStatsHelper: 2ms
02-20 16:25:51.715 16622 19942 W TileUtils: Found com.android.settings.backup.BackupSettingsActivity for intent Intent { act=com.android.settings.action.SETTINGS pkg=com.android.settings } missing metadata com.android.settings.category
02-20 16:25:51.722 16622 16645 I com.android.settings: android::hardware::configstore::V1_0::ISurfaceFlingerConfigs::hasWideColorDisplay retrieved: 0
02-20 16:25:51.725 16622 16645 I OpenGLRenderer: Initialized EGL, version 1.4
02-20 16:25:51.726 16622 16645 D OpenGLRenderer: Swap behavior 2
02-20 16:25:51.743 16622 16645 D mali_winsys: EGLint new_window_surface(egl_winsys_display *, void *, EGLSurface, EGLConfig, egl_winsys_surface **, EGLBoolean) returns 0x3000
02-20 16:25:51.756 16622 16645 I [Gralloc]: Got handle 1 for fd 70
02-20 16:25:51.756 16622 16645 I [Gralloc]: leave, w : 480, h : 480, format : 0x1,internal_format : 0x1, usage : 0xb00. size=921600,pixel_stride=480,byte_stride=1920
02-20 16:25:51.756 16622 16645 I [Gralloc]: leave: prime_fd=70,share_attr_fd=72
02-20 16:25:51.769 16622 19943 D Settings: No enabled state changed, skipping updateCategory call
02-20 16:25:51.789 16622 16639 D BatteryInfo: time for getStats: 75ms
02-20 16:25:51.802 16622 16622 D DashboardSummary: Listening for condition changes
02-20 16:25:51.802 16622 16622 D DashboardSummary: onConditionsChanged
02-20 16:25:51.812 16622 16622 D DashboardAdapter: adapter setConditions called
02-20 16:25:51.826 16622 16622 D DashboardSummary: conditions refreshed
02-20 16:25:51.827 16622 16622 D APM_Condition: APM condition refreshed
02-20 16:25:51.829 16622 16622 D APM_Condition: setActive was called with false
02-20 16:25:51.850 16622 19944 D BatteryInfo: time for regular BatteryInfo: 50ms
02-20 16:25:51.852 16622 19944 D BatteryInfo: time for getBatteryInfo: 2ms
02-20 16:25:51.861 16622 19944 D SuggestionParser: Day 326 for com.android.settings.suggested.category.FIRST_IMPRESSION
02-20 16:25:51.866 16622 16622 D BatteryInfo: time for callback: 1ms
02-20 16:25:51.868 16622 16645 I [Gralloc]: Got handle 2 for fd 75
02-20 16:25:51.868 16622 16645 I [Gralloc]: leave, w : 480, h : 480, format : 0x1,internal_format : 0x1, usage : 0xb00. size=921600,pixel_stride=480,byte_stride=1920
02-20 16:25:51.868 16622 16645 I [Gralloc]: leave: prime_fd=75,share_attr_fd=76
02-20 16:25:51.873 16622 19944 I SuggestionParser: Vingerafdruk gebruiken requires unavailable feature android.hardware.fingerprint
02-20 16:25:51.879 16622 19944 I SuggestionParser: Nog een vingerafdruk toevoegen requires unavailable feature android.hardware.fingerprint
02-20 16:25:53.320 16622 16645 I [Gralloc]: Got handle 3 for fd 78
02-20 16:25:53.320 16622 16645 I [Gralloc]: leave, w : 480, h : 480, format : 0x1,internal_format : 0x1, usage : 0xb00. size=921600,pixel_stride=480,byte_stride=1920
02-20 16:25:53.320 16622 16645 I [Gralloc]: leave: prime_fd=78,share_attr_fd=79
02-20 16:25:53.350 16622 16645 I [Gralloc]: Got handle 4 for fd 81
02-20 16:25:53.350 16622 16645 I [Gralloc]: leave, w : 480, h : 480, format : 0x1,internal_format : 0x1, usage : 0xb00. size=921600,pixel_stride=480,byte_stride=1920
02-20 16:25:53.350 16622 16645 I [Gralloc]: leave: prime_fd=81,share_attr_fd=82
02-20 16:25:53.855 16622 16622 I zygote64: Deoptimizing void android.widget.TextView.<init>(android.content.Context, android.util.AttributeSet, int, int) due to JIT inline cache
02-20 16:25:53.872 16622 16627 I zygote64: Do partial code cache collection, code=123KB, data=73KB
02-20 16:25:53.872 16622 16627 I zygote64: After code cache collection, code=83KB, data=55KB
02-20 16:25:53.873 16622 16627 I zygote64: Increasing code cache capacity to 512KB
02-20 16:25:54.017 16622 16622 D PowerUsageSummary: NO dashboard tiles for PowerUsageSummary
02-20 16:25:54.027 16622 19943 D SharedPreferencesLogger: Tried to log string preference fuelgauge.PowerUsageSummary/screen_timeout_battery = 1800000
02-20 16:25:54.048 16622 16622 I zygote64: Deoptimizing java.lang.Object de.robv.android.xposed.XposedBridge.handleHookedMethod(java.lang.reflect.Member, int, java.lang.Object, java.lang.Object, java.lang.Object[]) due to JIT inline cache
02-20 16:25:54.156 16622 16622 I zygote64: Deoptimizing int android.support.v7.widget.RecyclerView$5.getChildCount() due to JIT inline cache
02-20 16:25:54.162 16622 16627 I zygote64: Compiler allocated 8MB to compile void android.widget.TextView.<init>(android.content.Context, android.util.AttributeSet, int, int)
02-20 16:25:54.162 16622 16622 I zygote64: Deoptimizing int android.support.v7.widget.ChildHelper.getChildCount() due to JIT inline cache
02-20 16:25:54.208 16622 16622 D DashboardSummary: Stopped listening for condition changes
02-20 16:25:54.230 16622 19942 D BatteryInfoLoader: BatteryInfoLoader post query: 58ms
02-20 16:25:54.231 16622 19942 D BatteryInfo: time for getBatteryInfo: 1ms
02-20 16:25:54.231 16622 19942 D BatteryInfoLoader: BatteryInfoLoader.loadInBackground: 59ms
02-20 16:25:54.236 16622 19943 W TileUtils: Found com.android.settings.backup.BackupSettingsActivity for intent Intent { act=com.android.settings.action.SETTINGS pkg=com.android.settings } missing metadata com.android.settings.category
02-20 16:25:54.271 16622 16645 D mali_winsys: EGLint new_window_surface(egl_winsys_display *, void *, EGLSurface, EGLConfig, egl_winsys_surface **, EGLBoolean) returns 0x3000
02-20 16:25:54.299 16622 19943 D Settings: No enabled state changed, skipping updateCategory call
02-20 16:25:54.364 16622 16622 I zygote64: Deoptimizing int android.support.v7.widget.ChildHelper.getOffset(int) due to JIT inline cache
02-20 16:25:54.365 16622 16622 I zygote64: Deoptimizing android.view.View android.support.v7.widget.ChildHelper.getChildAt(int) due to JIT inline cache
02-20 16:25:54.367 16622 16622 I zygote64: Deoptimizing android.view.View android.support.v7.widget.RecyclerView$5.getChildAt(int) due to JIT inline cache
02-20 16:25:54.370 16622 16622 I zygote64: Deoptimizing android.view.View android.support.v7.widget.ViewBoundsCheck.findOneViewWithinBoundFlags(int, int, int, int) due to JIT inline cache
02-20 16:25:54.371 16622 16622 I zygote64: Deoptimizing int android.support.v7.widget.RecyclerView$LayoutManager$2.getChildStart(android.view.View) due to JIT inline cache
02-20 16:25:54.371 16622 16622 I zygote64: Deoptimizing int android.support.v7.widget.RecyclerView$LayoutManager.getDecoratedTop(android.view.View) due to JIT inline cache
02-20 16:25:54.371 16622 16622 I zygote64: Deoptimizing int android.support.v7.widget.RecyclerView$LayoutManager$2.getChildEnd(android.view.View) due to JIT inline cache
02-20 16:25:54.372 16622 16622 I zygote64: Deoptimizing int android.support.v7.widget.RecyclerView$LayoutManager.getDecoratedBottom(android.view.View) due to JIT inline cache
02-20 16:25:54.374 16622 16622 I zygote64: Deoptimizing int android.support.v7.widget.ScrollbarHelper.computeScrollOffset(android.support.v7.widget.RecyclerView$State, android.support.v7.widget.OrientationHelper, android.view.View, android.view.View, android.support.v7.widget.RecyclerView$LayoutManager, boolean, boolean) due to JIT inline cache
02-20 16:25:54.374 16622 16622 I zygote64: Deoptimizing int android.support.v7.widget.RecyclerView$LayoutManager.getPosition(android.view.View) due to JIT inline cache
02-20 16:25:54.380 16622 16633 I zygote64: Background concurrent copying GC freed 10014(795KB) AllocSpace objects, 5(348KB) LOS objects, 49% free, 4MB/8MB, paused 140us total 117.248ms
02-20 16:25:54.430 16622 16645 I [Gralloc]: Got handle 5 for fd 87
02-20 16:25:54.430 16622 16645 I [Gralloc]: leave, w : 480, h : 480, format : 0x1,internal_format : 0x1, usage : 0xb00. size=921600,pixel_stride=480,byte_stride=1920
02-20 16:25:54.430 16622 16645 I [Gralloc]: leave: prime_fd=87,share_attr_fd=88
02-20 16:25:54.616 16622 19943 D BatteryInfoLoader: BatteryInfoLoader post query: 8ms
02-20 16:25:54.632 16622 19943 D BatteryInfo: time for getBatteryInfo: 4ms
02-20 16:25:54.632 16622 19943 D BatteryInfoLoader: BatteryInfoLoader.loadInBackground: 24ms
02-20 16:25:54.831 16622 16645 I [Gralloc]: Got handle 6 for fd 90
02-20 16:25:54.831 16622 16645 I [Gralloc]: leave, w : 480, h : 480, format : 0x1,internal_format : 0x1, usage : 0xb00. size=921600,pixel_stride=480,byte_stride=1920
02-20 16:25:54.831 16622 16645 I [Gralloc]: leave: prime_fd=90,share_attr_fd=91
02-20 16:25:54.856 16622 16622 D AndroidRuntime: Shutting down VM
--------- beginning of crash
02-20 16:25:54.857 16622 16622 E AndroidRuntime: FATAL EXCEPTION: main
02-20 16:25:54.857 16622 16622 E AndroidRuntime: Process: com.android.settings, PID: 16622
02-20 16:25:54.857 16622 16622 E AndroidRuntime: java.lang.ArrayIndexOutOfBoundsException: length=7; index=7
02-20 16:25:54.857 16622 16622 E AndroidRuntime: 	at android.support.v7.preference.ListPreference.getEntry(ListPreference.java:228)
02-20 16:25:54.857 16622 16622 E AndroidRuntime: 	at android.support.v7.preference.ListPreference.getSummary(ListPreference.java:173)
02-20 16:25:54.857 16622 16622 E AndroidRuntime: 	at android.support.v7.preference.PreferenceManager$SimplePreferenceComparisonCallback.arePreferenceContentsTheSame(PreferenceManager.java:655)
02-20 16:25:54.857 16622 16622 E AndroidRuntime: 	at android.support.v7.preference.PreferenceGroupAdapter$2.areContentsTheSame(PreferenceGroupAdapter.java:183)
02-20 16:25:54.857 16622 16622 E AndroidRuntime: 	at android.support.v7.util.DiffUtil$DiffResult.findMatchingItems(DiffUtil.java:540)
02-20 16:25:54.857 16622 16622 E AndroidRuntime: 	at android.support.v7.util.DiffUtil$DiffResult.<init>(DiffUtil.java:484)
02-20 16:25:54.857 16622 16622 E AndroidRuntime: 	at android.support.v7.util.DiffUtil.calculateDiff(DiffUtil.java:184)
02-20 16:25:54.857 16622 16622 E AndroidRuntime: 	at android.support.v7.util.DiffUtil.calculateDiff(DiffUtil.java:92)
02-20 16:25:54.857 16622 16622 E AndroidRuntime: 	at android.support.v7.preference.PreferenceGroupAdapter.syncMyPreferences(PreferenceGroupAdapter.java:163)
02-20 16:25:54.857 16622 16622 E AndroidRuntime: 	at android.support.v7.preference.PreferenceGroupAdapter.-wrap0(Unknown Source:0)
02-20 16:25:54.857 16622 16622 E AndroidRuntime: 	at android.support.v7.preference.PreferenceGroupAdapter$1.run(PreferenceGroupAdapter.java:81)
02-20 16:25:54.857 16622 16622 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:790)
02-20 16:25:54.857 16622 16622 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:99)
02-20 16:25:54.857 16622 16622 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:164)
02-20 16:25:54.857 16622 16622 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6494)
02-20 16:25:54.857 16622 16622 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
02-20 16:25:54.857 16622 16622 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
02-20 16:25:54.857 16622 16622 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
02-20 16:25:54.857 16622 16622 E AndroidRuntime: 	at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:108)
02-20 16:25:54.880 16622 16622 I Process : Sending signal. PID: 16622 SIG: 9

**Comment by seaky** [comment]
please post the result of this command here:
```adb shell settings get system screen_off_timeout```

i think you somehow manually or with other 3rdparty tool, configured the sleep timeout.

**Reply by brewmark27** [reply]
Sorry for the delayed reply. The result is 1800000. I have no third party app that configures a sleep timeout, and if set manually, I have no idea where ...

**Reply by seaky** [reply]
Hi,

It should be ok but can you try this:
```adb shell settings put system screen_off_timeout 15000```

It will set your screen off timeout to 15sec
